lib/include/public/PauseManager.hpp

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2015-2020 Microsoft Corporation and Contributors.
+// SPDX-License-Identifier: Apache-2.0
+//
+#ifndef PAUSEMANAGER_HPP
+#define PAUSEMANAGER_HPP
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+
+#include "ctmacros.hpp"
+
+namespace MAT_NS_BEGIN
+{
+    class PauseManager {
+        public:
+
+        static void PauseActivity() noexcept;
```

Comment:
I think some of it can be implemented inside PAL layer. With unique OS-specific callback that monitors lifecycle events from Java or Obj-C layer. Then there is no need per se to have a separate public API. Also the corresponding "global app state" methods, such as `Suspend` + `Resume` / or `OnSuspending` + `OnResuming` can be added to existing `LogManager`. That makes it easier to project by adding methods  without adding a totally new class.

Reply:
I could see adding this as instance methods to the ILogManager interface, and make the pause/resume a per-instance decision. I could add this to the macro-instantiated static log managers. Either way, the API change does need to be represented in the Java/ObjC wrapppers.